pkg/reconciler/revision/resources/imagecache.go

```diff
@@ -28,15 +28,26 @@ import (
 )
 
 // MakeImageCache makes an caching.Image resources from a revision.
-func MakeImageCache(rev *v1alpha1.Revision) *caching.Image {
-	image := rev.Status.ImageDigest
+func MakeImageCache(rev *v1alpha1.Revision, containerName string) *caching.Image {
+	var image string
+	if len(rev.Spec.Containers) == 1 {
+		image = rev.Status.ImageDigest
+	} else if len(rev.Spec.Containers) > 1 {
+		for i := range rev.Spec.Containers {
+			if len(rev.Spec.Containers[i].Ports) != 0 {
```

Comment:
This lacks a check against `containerName`

Reply:
I think we don't get `containerName` as empty because its handled in [validation](https://github.com/knative/serving/pull/6739/commits/5e8e969160bc6c63639369eebcddf981667d6587#diff-daca68b8123c709e8127c6c834c5c023R56)